lib/configuration.js

```diff
@@ -256,14 +332,17 @@ function applyConfiguration(inConfig) {
     );
   }
 
-  return outConfig;
+  return resolvedConfig;
 }
 
-
 module.exports = {
   applyConfiguration,
   applyLoggingOptions,
 
   // only for the purpose of unit tests
-  _coerceRemovedOptions: coerceRemovedOptions,
+  _normalizeConfig: normalizeConfig,
+  _validateConfig: validateConfig,
+  _resolveConfig: resolveConfig,
+  _coerceRemovedOptions: coerceDeprecatedOptions,
+  _coerceDeprecatedLevel: coerceDeprecatedLevelOption,
```

Comment:
Shouldn't we rename these?

Reply:
Will do. Also will add those functions that are subjected to unit tests.